packages/ai-constructs/src/conversation/conversation_handler_construct.ts

```diff
@@ -0,0 +1,104 @@
+import { FunctionResources, ResourceProvider } from '@aws-amplify/plugin-types';
+import { Duration } from 'aws-cdk-lib';
+import { Effect, PolicyStatement } from 'aws-cdk-lib/aws-iam';
+import { CfnFunction, Runtime as LambdaRuntime } from 'aws-cdk-lib/aws-lambda';
+import { NodejsFunction } from 'aws-cdk-lib/aws-lambda-nodejs';
+import {
+  CustomDataIdentifier,
+  DataProtectionPolicy,
+  LogGroup,
+  RetentionDays,
+} from 'aws-cdk-lib/aws-logs';
+import { Construct } from 'constructs';
+import path from 'path';
+
+const resourcesRoot = path.normalize(path.join(__dirname, 'runtime'));
+const defaultHandlerFilePath = path.join(resourcesRoot, 'default_handler.js');
+
+export type ConversationHandlerFunctionProps = {
+  entry?: string;
+  models: Array<{
+    modelId: string;
+    region: string;
+  }>;
+};
+
+/**
+ * Conversation Handler Function CDK construct.
+ * This construct deploys resources that integrate conversation routes
+ * defined in data schema with AI models available in AWS Bedrock. I.e.
+ * 1. AWS Lambda function that handles conversation turn events.
+ *    With Amplify provided implementation by default and option to specify
+ *    custom handler.
+ * 2. AWS CloudWatch log group policy with appropriate data protection policies.
+ * 3. AWS IAM policy that grants access to selected AWS Bedrock models.
+ */
+export class ConversationHandlerFunction
+  extends Construct
+  implements ResourceProvider<FunctionResources>
+{
+  resources: FunctionResources;
+
+  /**
+   * Creates Conversation Handler Function CDK construct.
+   */
+  constructor(
+    scope: Construct,
+    id: string,
+    private readonly props: ConversationHandlerFunctionProps
+  ) {
+    super(scope, id);
+
+    if (this.props.entry && !path.isAbsolute(this.props.entry)) {
+      throw new Error('Entry must be absolute path');
+    }
```

Comment:
I added this restriction for now. We need some smarter resolution logic which is similar to what `defineFunction` and `NodeJSFunction` construct is doing. But I didn't want to inflate this PR with that.
We'll address this later (when `defineConversationHandler`) happens.